packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -189,6 +197,47 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
     }
 
+    /**
+     * @notice sell long wPowerPerp and flashswap mint short position
+     * @dev flahswap amount = collateral amount - msg.value - ETH from selling long wPowerPerp
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmountToMint wPowerPerp amount to mint
+     * @param _collateralAmount collateral amount to use for minting
+     * @param _wPowerPerpAmountToSell long wPowerPerp amount to sell
+     * @param _minToReceive min ETH amount to receive for selling long wPowerPerp amount
+     */
+    function flashswapSellLongWMint(
+        uint256 _vaultId,
+        uint256 _wPowerPerpAmountToMint,
+        uint256 _collateralAmount,
+        uint256 _wPowerPerpAmountToSell,
+        uint256 _minToReceive
+    ) external payable {
+        IWPowerPerp(wPowerPerp).transferFrom(msg.sender, address(this), _wPowerPerpAmountToSell);
+
+        // swap long wPowerPerp
+        uint256 amountOut = _exactInFlashSwap(
+            wPowerPerp,
+            weth,
+            IUniswapV3Pool(wPowerPerpPool).fee(),
+            _wPowerPerpAmountToSell,
+            _minToReceive,
+            uint8(FLASH_SOURCE.SWAP),
+            ""
+        );
+
+        // flahswap and mint short position
+        _exactInFlashSwap(
+            wPowerPerp,
+            weth,
+            IUniswapV3Pool(wPowerPerpPool).fee(),
+            _wPowerPerpAmountToMint,
+            _collateralAmount.sub(msg.value).sub(amountOut),
```

Comment:
don't think this is correct/works

ex: if the user is selling a lot of oSQTH (which gives a large amountOut), and then only shorting a small amount, it could revert

think the above refactor would make this more clear

Reply:
a user may be due a refund from this tx (they sell $1000 long oSQTH and then short $1 more w $2 of collateral)